lasagne/layers/special.py

```diff
@@ -354,3 +356,63 @@ def _meshgrid(height, width):
     ones = T.ones_like(x_t_flat)
     grid = T.concatenate([x_t_flat, y_t_flat, ones], axis=0)
     return grid
+
+
+class ParametricRectifierLayer(Layer):
+    """
+    lasagne.layers.ParametricRectifierLayer(incoming,
+    alpha=init.Uniform((0.001, 0.1)), **kwargs)
+
+    A layer that applies parametric rectify nonlinearity to its input
+    following K He, X Zhang et al. (http://arxiv.org/abs/1502.01852):
+    :math:`\\varphi(y_i) = \\max(x_i,0) + \\alpha_i \\min(x_i,0)`
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    a : Theano shared variable, numpy array or callable
+        An initializer for the weights of the layer. If a shared variable or a
+        numpy array is provided the shape should  be incoming.shape[1:]).
+        See :meth:`Layer.create_param` for more information.
+    """
+    def __init__(self, incoming, alpha=init.Uniform((0.001, 0.1)), axes=None,
+                 **kwargs):
+        super(ParametricRectifierLayer, self).__init__(incoming, **kwargs)
+        if axes is None:
+            # default: the parameters are same across only second axis
+            # this will do correct thing for both dense operations
+            # and convolution
+            axes = (1, )
+        elif isinstance(axes, int):
+            axes = (axes,)
+        shape = [1, ] * len(self.input_shape)
+        for axis in axes:
+            shape[axis] = self.input_shape[axis]
+        self.broadcast = tuple(set(range(len(self.input_shape))) - set(axes))
```

Comment:
I'd prefer to store `self.axes` (because that's what was given in the constructor) and handle the broadcasting in `get_output_for()`. I think I'd also prefer `alpha` to have fewer dimensions and add them via `dimshuffle` instead of setting them to `1` and adding them via broadcasting. See `BiasLayer` or this prior implementation of a PReLULayer: https://github.com/Lasagne/Lasagne/pull/247/files#diff-bce2cd455789587cf49cbb4b404b5a3dR15
I like the idea of having `axes` specify the axes to untie over, rather than the axes to share over. This seems easier.